doc/index.rst

```diff
@@ -193,6 +193,7 @@ PennyLane is **free** and **open source**, released under the Apache License, Ve
 
    development/guide
    development/plugins
+   development/adding_templates
    development/release_notes.md
 
 .. toctree::
```

Comment:
(Closest I could get with the comment): down below in `doc/introduction/templates.rst` a `Developer's guide` -> `Development guide` renaming would be needed.

Reply:
Good catch! I've updated it now to reference to the correct page.

Regarding the warnings, it's because the new sub-pages aren't in the TOC. I'm not sure how to best solve this though without adding them to the TOC (maybe they can be there but hidden somehow).